src/lib/io/sol-memmap-storage.c

```diff
@@ -513,3 +753,47 @@ sol_memmap_remove_map(const struct sol_memmap_map *map)
 
     return sol_ptr_vector_remove(&memory_maps, map);
 }
+
+SOL_API bool
+sol_memmap_set_timeout(struct sol_memmap_map *map, unsigned int timeout)
+{
+    struct sol_memmap_map *iter;
+    int i;
+
+    SOL_NULL_CHECK(map, false);
+
+    /* Rememeber, as we may have a copy of map (due to device resolving),
+     * we need to update our proper copy */
+    SOL_PTR_VECTOR_FOREACH_IDX (&memory_maps, iter, i) {
+        if (iter->entries == map->entries) {
+            map->timeout = timeout;
```

Comment:
what if this is not a `resolved_map`? Then you're writing to a read-only memory and it will crash if proper protection is set.

I think a good approach to simplify this is to create a new internal structure that extends the `sol_memmap_map` with the timeout and always use from there. This could serve the purpose of resolve the path as well, then you keep a single structure and simplify other paths.